Nx Nextjs build produce broken dist app MODULE_NOT_FOUND

<!-- Please do your best to fill out all of the sections below! -->

## Current Behavior
Building a nextjs application for production creates a deployable application that cannot start due to the following error:
`Error: Cannot find module '@nrwl/next/plugins/with-nx'`

When deploying the application as SSR app in a docker image, it would not make sense to add all the devDependencies.

## Expected Behavior
Either of the following two:
- the produced nextjs application next.config.js does not have any dependency on  Nx plugin (`@nrwl/next/plugins/with-nx`)
- the dependent plugin `@nrwl/next/plugins/with-nx` is added as dependency for production (not as a devDependency)

## Steps to Reproduce
<!-- Help us help you by making it easy for us to reproduce your issue! -->
Based on the nx official docs:

```shell
npx create-nx-workspace@latest test-workspace
cd test-workspace
yarn
yarn add -D @nrwl/next
nx generate @nrwl/next:application test-app
# styled-components
yarn nx build test-app --prod

# remove node_modules since it won't be present in prod 
# yarn install will be run in the built application
rm -rf node_modules

# Prepare app for prod deployment
cd  dist/apps/test-app
yarn install --production=true
NODE_ENV=production yarn start
```

### Failure Logs
```shell
NODE_ENV=production yarn start
yarn run v1.22.10
warning package.json: No license field
$ next start
ready - started server on 0.0.0.0:3000, url: http://localhost:3000
Error: Cannot find module '@nrwl/next/plugins/with-nx'
Require stack:
- /home/workspaces/tmp/test-workspace/dist/apps/test-app/next.config.js
- /home/workspaces/tmp/test-workspace/dist/apps/test-app/node_modules/next/dist/next-server/server/config.js
- /home/workspaces/tmp/test-workspace/dist/apps/test-app/node_modules/next/dist/next-server/server/next-server.js
- /home/workspaces/tmp/test-workspace/dist/apps/test-app/node_modules/next/dist/server/next.js
- /home/workspaces/tmp/test-workspace/dist/apps/test-app/node_modules/next/dist/server/lib/start-server.js
- /home/workspaces/tmp/test-workspace/dist/apps/test-app/node_modules/next/dist/cli/next-start.js
- /home/workspaces/tmp/test-workspace/dist/apps/test-app/node_modules/next/dist/bin/next
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:815:15)
    at Function.mod._resolveFilename (/home/workspaces/tmp/test-workspace/dist/apps/test-app/node_modules/next/dist/build/webpack/require-hook.js:4:1784)
    at Function.Module._load (internal/modules/cjs/loader.js:667:27)
    at Module.require (internal/modules/cjs/loader.js:887:19)
    at require (internal/modules/cjs/helpers.js:74:18)
    at Object.<anonymous> (/home/workspaces/tmp/test-workspace/dist/apps/test-app/next.config.js:2:16)
    at Module._compile (internal/modules/cjs/loader.js:999:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1027:10)
    at Module.load (internal/modules/cjs/loader.js:863:32)
    at Function.Module._load (internal/modules/cjs/loader.js:708:14) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/home/workspaces/tmp/test-workspace/dist/apps/test-app/next.config.js',
    '/home/workspaces/tmp/test-workspace/dist/apps/test-app/node_modules/next/dist/next-server/server/config.js',
    '/home/workspaces/tmp/test-workspace/dist/apps/test-app/node_modules/next/dist/next-server/server/next-server.js',
    '/home/workspaces/tmp/test-workspace/dist/apps/test-app/node_modules/next/dist/server/next.js',
    '/home/workspaces/tmp/test-workspace/dist/apps/test-app/node_modules/next/dist/server/lib/start-server.js',
    '/home/workspaces/tmp/test-workspace/dist/apps/test-app/node_modules/next/dist/cli/next-start.js',
    '/home/workspaces/tmp/test-workspace/dist/apps/test-app/node_modules/next/dist/bin/next'
  ]
}
error Command failed with exit code 1.
```

### Environment
<!-- It's important for us to know the context in which you experience this behavior! -->
<!-- Please paste the result of `nx report` below! -->
nx won't be installed in the docker image nor in the dist folder and this is actually the problem.

All the commands have been run with the latest nx version:
```shell
  Node : 12.21.0
  OS   : linux x64
  npm  : 7.10.0
  
  nx : Not Found
  @nrwl/angular : Not Found
  @nrwl/cli : 12.0.8
  @nrwl/cypress : 12.0.8
  @nrwl/devkit : 12.0.8
  @nrwl/eslint-plugin-nx : 12.0.8
  @nrwl/express : Not Found
  @nrwl/jest : 12.0.8
  @nrwl/linter : 12.0.8
  @nrwl/nest : Not Found
  @nrwl/next : 12.0.8
  @nrwl/node : Not Found
  @nrwl/react : 12.0.8
  @nrwl/schematics : Not Found
  @nrwl/tao : 12.0.8
  @nrwl/web : 12.0.8
  @nrwl/workspace : 12.0.8
  @nrwl/storybook : 12.0.8
  @nrwl/gatsby : Not Found
  typescript : 4.2.4
```

--- a/packages/next/src/generators/init/init.spec.ts
+++ b/packages/next/src/generators/init/init.spec.ts
@@ -13,8 +13,8 @@ describe('init', () => {
   it('should add react dependencies', async () => {
     await nextInitGenerator(tree, {});
     const packageJson = readJson(tree, 'package.json');
-    expect(packageJson.dependencies['@nrwl/next']).toBeUndefined();
     expect(packageJson.dependencies['@nrwl/react']).toBeUndefined();
+    expect(packageJson.dependencies['@nrwl/next']).toBeDefined();
     expect(packageJson.dependencies['next']).toBeDefined();
   });
 

--- a/packages/next/src/generators/init/init.ts
+++ b/packages/next/src/generators/init/init.ts
@@ -25,14 +25,13 @@ function updateDependencies(host: Tree) {
   return addDependenciesToPackageJson(
     host,
     {
+      '@nrwl/next': nxVersion,
       next: nextVersion,
       react: reactVersion,
       'react-dom': reactDomVersion,
       tslib: '^2.0.0',
     },
-    {
-      '@nrwl/next': nxVersion,
-    }
+    {}
   );
 }
 

--- a/packages/workspace/src/generators/new/new.ts
+++ b/packages/workspace/src/generators/new/new.ts
@@ -262,10 +262,10 @@ const presetDependencies: Omit<
     },
   },
   [Preset.NextJs]: {
-    dependencies: {},
-    dev: {
+    dependencies: {
       '@nrwl/next': nxVersion,
     },
+    dev: {},
   },
   [Preset.Gatsby]: {
     dependencies: {},

